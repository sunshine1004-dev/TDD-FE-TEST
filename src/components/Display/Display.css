.display-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 0 4%;
}

.display-value {
  font-family: var(--main-font);
  margin-left: auto;
  overflow: hidden;
}
