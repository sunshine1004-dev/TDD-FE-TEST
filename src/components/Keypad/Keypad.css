.keypad-container {
  display: flex;
  flex-direction: row;
  padding: 2%;
  margin-top: 0.5rem;
}
.argument-input {
  box-shadow: 0 0 0.25rem rgba(0, 0, 0, 0.15);
  border: 0.125rem solid #cccbcd;
  border-radius: 0.5rem;
  color: #3d3c3e;
}

.argument-input:focus {
  border-color: #5f2ad0;
  outline: none;
  caret-color: #5f2ad0;
}

.submit-button {
  cursor: pointer;
  border-radius: 0.5rem;
  border: 0.125rem solid currentColor;
  background: #fff;
  color: #124ce1cc;
  height: 2.2rem;
  width: 7rem;

  margin-left: 0.5rem;
}

.submit-button:hover {
  transition: all 0.2s ease 0s;
  box-shadow: rgb(4 17 29 / 25%) 0px 0px 8px 0px;
  background: #124ce1cc;
  color: #fff;
}
