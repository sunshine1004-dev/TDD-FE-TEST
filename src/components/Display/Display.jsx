import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ sum, average, deviation }) => (
    <div className="display-container">
        <p className="display-value">sum: {sum}</p>
        <p className="display-value">average: {average}</p>
        <p className="display-value">deviation: {deviation}</p>
    </div>
);

Display.propTypes = {
    sum: PropTypes.string.isRequired,
    average: PropTypes.string.isRequired,
    deviation: PropTypes.string.isRequired,
};

Display.defaultProps = { sum: '0', average: '0', deviation: '0' };

export default Display;
