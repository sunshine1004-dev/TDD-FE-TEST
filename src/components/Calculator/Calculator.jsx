import React, { Component } from 'react';
import axios from 'axios';
import './Calculator.css';

import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';
import './Calculator.css';

class Calculator extends Component {
    state = {
        sum: '0',
        average: '0',
        deviation: '0',
        argument: '',
    };

    submit = () => {
        const that = this;
        const { argument } = that.state;
        if (argument !== '') {
            axios
                .post('http://localhost:3001/api/calculate', {
                    argument,
                })
                .then(function (response) {
                    const resp = response.data;
                    that.setState({
                        sum: resp.sum.toString(),
                        average: resp.average.toFixed(2).toString(),
                        deviation: resp.deviation.toFixed(2).toString(),
                    });
                })
                .catch(function (error) {
                    console.log(error);
                });
        }
    };

    setArguments = (value) => {
        const regex = /[^\d.,]|\.(?=.*\.)|\,(?=,*\,)/g;
        const subst = ``;

        const result = value.replace(regex, subst);
        this.setState({ argument: result });
        this.setState({
            sum: '0',
            average: '0',
            deviation: '0',
        });
    };

    render() {
        const { argument, sum, average, deviation } = this.state;

        return (
            <div className="calculator-container">
                <Display sum={sum} average={average} deviation={deviation} />
                <Keypad argument={argument} handleSubmit={this.submit} setArguments={this.setArguments} />
            </div>
        );
    }
}

export default Calculator;
