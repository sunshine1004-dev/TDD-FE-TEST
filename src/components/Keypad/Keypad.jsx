import React from 'react';
import PropTypes from 'prop-types';
import './Keypad.css';

const Keypad = ({ argument, setArguments, handleSubmit }) => {
    return (
        <div className="keypad-container">
            <input
                className="argument-input"
                autoFocus
                onChange={(e) => setArguments(e.target.value)}
                value={argument}
                placeholder="1,3,5"
            />
            <button className="submit-button" onClick={handleSubmit}>
                submit
            </button>
        </div>
    );
};

Keypad.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    setArguments: PropTypes.func.isRequired,
};

export default Keypad;
