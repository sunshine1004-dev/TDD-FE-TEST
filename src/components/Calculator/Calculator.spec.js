import React from 'react';
import { mount, shallow } from 'enzyme';
import Calculator from './Calculator';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

describe('Calculator', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Calculator />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the Display and Keypad Components', () => {
    expect(
      wrapper.containsAllMatchingElements([
        <Display
          sum={wrapper.instance().state.sum}
          average={wrapper.instance().state.average}
          deviation={wrapper.instance().state.deviation}
        />,
        <Keypad
          argument={wrapper.instance().state.argument}
          handleSubmit={wrapper.instance().submit}
          setArguments={wrapper.instance().setArguments}
        />,
      ])
    ).toEqual(true);
  });

  describe('submit', () => {
    it('submit argument', () => {
      wrapper.instance().submit('5');
    });
  });

  describe('setArguments', () => {
    it('updates the state of argument', () => {
      wrapper.instance().setArguments('1,3,5');
      expect(wrapper.state('argument')).toEqual('1,3,5');
    });
  });
});
