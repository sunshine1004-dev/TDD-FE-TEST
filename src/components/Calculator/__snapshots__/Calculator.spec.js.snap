// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly 1`] = `
<div
  className="calculator-container"
>
  <Display
    average="0"
    deviation="0"
    sum="0"
  />
  <Keypad
    argument=""
    handleSubmit={[Function]}
    setArguments={[Function]}
  />
</div>
`;
